src/dynamics/solver/parallel_island_solver.rs

```diff
@@ -229,6 +230,16 @@ impl ParallelIslandSolver {
                     parallel_joint_constraints.constraint_descs.len(),
                 );
 
+                // Symplectic Euler: move bodies using the *old* velocities.
+                concurrent_loop! {
+                    let batch_size = thread.batch_size;
+                    for handle in active_bodies[thread.body_integration_index, thread.num_integrated_bodies] {
+                        let rb = &mut bodies[handle.0];
+                        rb.integrate(params.dt);
+                        positions[rb.active_set_offset] = rb.position;
+                    }
+                }
+
```

Comment:
I am kind of winging it here with the concurrent version.

The `concurrent_loop` macro is a bit... undocumented :P